fetchai/ledger/contract.py

```diff
@@ -129,7 +129,7 @@ def action(self, api: ContractsApiLike, name: str, fee: int, signers: List[Entit
             resource_addresses = [ShardMask.state_to_address(address, self) for address in
                                   self._parser.used_globals_to_addresses(name, list(args))]
         except (UnparsableAddress, UseWildcardShardMask):
-            print("WARNING: Couldn't auto-detect used shards, using wildcard shard mask")
+            logging.warn("Couldn't auto-detect used shards, using wildcard shard mask")
```

Comment:
logging.warn is deprecated, use logging.warning instead.
```suggestion
            logging.warning("Couldn't auto-detect used shards, using wildcard shard mask")
```

Reply:
Done.